mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -1041,6 +1041,24 @@ class LogisticRegression @Since("1.2.0") (
     (solution, arrayBuilder.result())
   }
 
+  private[spark] override def estimateModelSize(dataset: Dataset[_]): Long = {
+    // TODO: get numClasses and numFeatures together from dataset
+    val numClasses = DatasetUtils.getNumClasses(dataset, $(labelCol))
+    val numFeatures = DatasetUtils.getNumFeatures(dataset, $(featuresCol))
+
+    var size = this.estimateMatadataSize
+    if (checkMultinomial(numClasses)) {
+      size += Matrices.getDenseSize(numFeatures, numClasses) // coefficientMatrix
+      size += Vectors.getDenseSize(numClasses) // interceptVector
+    } else {
+      size += Matrices.getDenseSize(numFeatures, 1) // coefficientMatrix
+      size += Vectors.getDenseSize(1) // interceptVector
+    }
+    size += java.lang.Integer.BYTES // numClasses
+    size += 1 // isMultinomial
```

Comment:
emm, do we need to estimate size including these tiny variables ?

seemingly it is not necessary (the estimation already has estimation error larger than this), e.g., introduced by java object metadata

Reply:
i am fine if we only focus on larger ones like vector/matrix/array